include/odp/api/spec/schedule.h

```diff
@@ -187,6 +207,41 @@ void odp_schedule_prefetch(int num);
  */
 int odp_schedule_num_prio(void);
 
+/**
+ * Initialize schedule configuration options
+ *
+ * Initialize an odp_schedule_config_t to its default values.
+ *
+ * @param[out] config  Pointer to schedule configuration structure
+ */
+void odp_schedule_config_init(odp_schedule_config_t *config);
+
+/**
+ * Global schedule configuration
+ *
+ * Initialize and configure scheduler with global configuration options.
+ *
+ * @param[in,out] config   Pointer to scheduler configuration structure
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ *
+ * @see odp_schedule_capability(), odp_schedule_config_init()
+ */
+int odp_schedule_config(odp_schedule_config_t *config);
```

Comment:
I think output param is needed only for implementation to tell if flow id space is power of two (if it rounded it up). Now, application that sets flow id itself would not need to know that (it would just use it's own range). The problem is only with packet input, if implementation calculates hash and sets that as flow id, application may see larger flow ids than it itself configured here.

So, would it be better to have this as input param and add a pktio capability/config to fix how many bits packet input will use as flow id?

Reply:
@psavol also we can just restrict that flow id space should always be of a power of two. What do you think?

Reply:
Actually would it work if implementation rounds flow IDs down to power of 2. E.g. max num flow id in capa is 128, application configures and uses itself only 100, implementation rounds down number of packet input flow IDs to 64.